src/vmm/src/device_manager/mmio.rs

```diff
@@ -249,8 +249,13 @@ impl MMIODeviceManager {
         &mut self,
         vm: &VmFd,
         serial: Arc<Mutex<devices::legacy::Serial>>,
+        dev_info_opt: Option<MMIODeviceInfo>,
     ) -> Result<()> {
-        let slot = self.allocate_new_slot(1)?;
+        let slot = if let Some(slot) = dev_info_opt {
+            slot
+        } else {
+            self.allocate_new_slot(1)?
+        };
```

Comment:
nit: shorter:
```suggestion
        let slot = dev_info_opt.unwrap_or(self.allocate_new_slot(1)?);
```
(same below)

Reply:
Done.